src/NLPModels.jl

```diff
@@ -272,6 +281,25 @@ Evaluate ``∇c(x)v``, the Jacobian-vector product at `x` in place.
 jprod!(::AbstractNLPModel, ::AbstractVector, ::AbstractVector, ::AbstractVector) =
   throw(NotImplementedError("jprod!"))
 
+"""
+    Jv = jprod!(nlp, rows, cols, vals, v, Jv)
+
+Evaluate ``∇c(x)v``, the Jacobian-vector product, where the Jacobian is given by
+`(rows, cols, vals)` in triplet format.
+"""
+function jprod!(nlp::AbstractNLPModel, rows::AbstractVector{<: Integer}, cols::AbstractVector{<: Integer}, vals::AbstractVector, v::AbstractVector, Jv::AbstractVector)
+  increment!(nlp, :neval_jprod)
+  coo_prod!(nlp.meta.ncon, nlp.meta.nnzj, rows, cols, vals, v, Jv)
+end
+
+"""
+    Jv = jprod!(nlp, x, rows, cols, v, Jv)
+
+Evaluate ``∇c(x)v``, the Jacobian-vector product at `x` in place.
+`(rows, cols)` should be the Jacobian structure in triplet format.
+"""
+jprod!(nlp::AbstractNLPModel, x::AbstractVector, ::AbstractVector{<: Integer}, ::AbstractVector{<: Integer}, v::AbstractVector, Jv::AbstractVector) = jprod!(nlp, x, v, Jv)
```

Comment:
Any reason why this is better than calling `jac_coord`?

Reply:
For `ADNLPModel`, for instance, `jprod!(nlp, x, v, Jv)` is defined cheaply, and `jac_coord` is much more expensive. Also for `CUTEstModel`, `jprod!` is defined using specialized CUTEst functions.